### PR TITLE
[reland] Add annotations to PRs from forks

### PR DESCRIPTION
**Pull request by samestep**
We've been using [pytorch/add-annotations-github-action](https://github.com/pytorch/add-annotations-github-action) to add annotations to PRs when they fail Flake8 or clang-tidy. Up until now, though, that functionality has only worked on PRs in pytorch/pytorch itself, not on PRs from forks. This PR fixes that using a technique from [this GitHub blog post](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) (also linked in a comment in this diff).

**Test plan:**

@janeyx99 and I tested this in the same GitHub repo used to test #54685 and #54693, including with PRs from forks.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -68,27 +68,25 @@ jobs:
...
       - name: Run flake8
         run: |
           set -eux
           pip install -r requirements-flake8.txt
           flake8 --version
-          flake8 | tee ${GITHUB_WORKSPACE}/flake8-output.txt
-      - name: Add annotations
-        uses: pytorch/add-annotations-github-action@master
+          flake8 | tee ${GITHUB_WORKSPACE}/flake8-output/warnings.txt
+      - name: Upload artifact
```
Comment (walterddr) [inline .github/workflows/lint.yml:82]: wondering did we measure the impact of these artifacts upload in terms of performance/overall CI time? clang-tidy can get huge

**Review by walterddr — APPROVED** [review]
looks good to me

File: .github/workflows/lint.yml (hunk 1 above)
Comment (samestep) [reply .github/workflows/lint.yml:82]: oh good question, I actually have no idea

**Comment by samestep** [comment]
Reverting because apparently this is running on `master`, which is not supposed to do.

**Pull request closed without merging**